File: py/evaluate_all_starting_points.py
import os
import sys

import numpy as np
import scipy as sp
from calfun import calfun
from dfoxs import dfoxs

# np.seterr("raise")


def doit():
    dfo = np.loadtxt("../data/dfo.dat")
    Results = {}

    for p, probtype in enumerate(["absnormal", "absuniform", "abswild", "noisy3", "nondiff", "relnormal", "reluniform", "relwild", "smooth", "wild3"]):
        for row, (nprob, n, m, factor_power) in enumerate(dfo[0:1]):
            n = int(n)
            m = int(m)

            X0 = dfoxs(n, nprob, int(10**factor_power)).T
            [y, F, G, J] = calfun(X0, m, int(nprob), probtype, gradout=True)

            Results[str(p) + "_" + str(row)] = {}
            Results[str(p) + "_" + str(row)]["y"] = y
            Results[str(p) + "_" + str(row)]["F"] = F
            Results[str(p) + "_" + str(row)]["G"] = G
            Results[str(p) + "_" + str(row)]["J"] = J


if __name__ == "__main__":
    doit()

File: m/evaluate_all_starting_points.m
load('../data/dfo.dat');
probtypes = ["absnormal", "absuniform", "abswild", "noisy3", "nondiff", "relnormal", "reluniform", "relwild", "smooth", "wild3"];

Results = cell(length(probtypes), size(dfo, 1));
for p = 1:length(probtypes)
    for row = 1:size(dfo, 1)
        nprob = dfo(row, 1);
        n = dfo(row, 2);
        m = dfo(row, 3);
        factor_power = dfo(row, 4);

        X0 = dfoxs(n, nprob, 10^factor_power);

        BenDFO.nprob = nprob;
        BenDFO.m = m;
        BenDFO.n = n;

        [y, F, G, J] = calfun(X0, BenDFO, probtypes(p));

        Results{p, row}.X0 = X0;
        Results{p, row}.y = y;
        Results{p, row}.F = F;
        Results{p, row}.G = G;
        Results{p, row}.J = J;
    end
end
save("fvec_and_gradients_at_starting_values_matlab.mat", "Results");
